test/strict.coffee

```diff
@@ -19,6 +19,9 @@
 # helper to assert that code complies with strict prohibitions
 strict = (code, msg) ->
   throws (-> CoffeeScript.compile code), null, msg ? code
+strictAst = (code, msg) ->
```

Comment:
> I think the original authors would instead make a new helper like `strictAst` that could have the same signature as `strict`

You're right that does seem in keeping with the existing style. Updated to use `strictAst()` here and `assertErrorMessageAst()` in `test/error_messages.coffee`